snakeV2.py

```diff
@@ -0,0 +1,353 @@
+            xfact = 1
+
+        for i in range(num):  # generate and add blocks to self.blocks list in proper direction
+            ypos = self.blocks[-1].y + BlockSize*i*yfact
+            xpos = self.blocks[-1].x + BlockSize*i*xfact
+            block = pygame.Rect(xpos, ypos, BlockSize, BlockSize)
+            self.blocks.append(block)
+
+    def check_collision(self):
+        """ Goes through each block in list and determines if head block is in same position as that body block.
+            If any block is in the same position as the head block the function returns true. If it goes through
+            the for loop without finding any collision if defaults to returning fales.
+        """
+        count = 0
+        for block in self.blocks:
+            if count != 0 and self.blocks[0].x == block.x and self.blocks[0].y == block.y:
+                return True
+            count += 1
+        return False
+
+    def check_boundary(self):
+        """ Determines if head block exits pygame screen window (if x or y position is < 0 or > window size)
+            Defaults to returning false.
+        """
+        x = self.blocks[0].x
+        y = self.blocks[0].y
+        if x < 0 or y < 0 or x >= WINDOW or y >= WINDOW:
+            return True
+        return False
+
+    def check_food(self, food):
+        """ Checks if head block is in the same x and y position as the food block. Takes the food block as an input argument
+            to use its x and y position.
+            Defaults to returning false.
+        """
+        if self.blocks[0].x == food.x and self.blocks[0].y == food.y:
+            return True
+        return False
+
+    def reset(self):
+        self.__init__()
+
+
+"""Defines a class that visualizes the snake in the gameplay window.
+"""
+
+
+class SnakeView(object):
+    def __init__(self, model):
+        self.model = model
+
+    def draw(self, surface):
+        """ For each block in the snake model blocks list SnakeView draws the rect object
+            as a blue rectangle which is part of the snake's body.
+        """
+        model = self.model
+        for rect in model.blocks:
+            pygame.draw.rect(surface, BLUE, rect)
+
+
+"""Defines a class that initializes the food objects in random locations in the
+window and can reset their locations.
+"""
+
+
+class Food(object):
+    """ The food model contains one rect object which is randomly assigned an x and y position.
+        The step function could be used for interesting game funcitonality later on but is
+        passed for now.
+    """
+    def __init__(self):
+        self.x = random.randint(0, WINDOW/BlockSize-1)*BlockSize
+        self.y = random.randint(0, WINDOW/BlockSize-1)*BlockSize
+        self.rect = pygame.Rect(self.x, self.y, BlockSize, BlockSize)
+
+    def step(self):
+        pass
+
+    def reset(self):
+        self.__init__()
+
+
+"""Visualizes the food objects in the gameplay window."""
+
+
+class FoodView(object):
```

Comment:
Maybe instead of splitting into separate files for each class, splitting into Model View Controller files, then organizing each class in that manner might make more sense